### PR TITLE
Add and update developer docs

### PR DESCRIPTION
**Pull request by corymhall**
Using codex to generate/update some docs.

### CODE REVIEW

**Comment by t0yv0** [comment]
@corymhall what are you looking for in terms of review? thoroughly or stamp? Can you highlight areas to double-check?

File: docs/guides/testing.md [hunk 1]
```diff
@@ -1,158 +1,216 @@
...
+`pkg/tfbridge/provider_test.go`). They are helpful when higher-level harnesses are impractical, but new coverage should
+use schema + program or cross-tests instead. Treat replays as stop-gaps and backfill more maintainable coverage later.
 
+## Golden Files & Accepting Changes
 
-## Property-Based Testing
+Some tests assert against generated output stored alongside fixtures. When behavior changes intentionally:
 
-Building on the cross-tests, we also have property based tests using the [Rapid](https://github.com/flyingmutant/rapid) library. These tests are useful for covering a wide variety of inputs and outputs. These are still somewhat experimental and found under [./pkg/internal/tests/cross-tests/rapid_test.go](https://github.com/pulumi/pulumi-terraform-bridge/blob/317d4b819f12d4bc66adc5fb248bfb77a3cc7ba7/pkg/internal/tests/cross-tests/rapid_test.go#L30).
+1. Run `make test_accept` (sets `PULUMI_ACCEPT=1`) to update goldens.
```
Comment (VenelinMartinov) [inline docs/guides/testing.md:206]: This isn't correct AFAIK - `autogold`, which is the library we use for most golden tests use `go test -update`

File: docs/guides/testing.md [hunk 2]
```diff
@@ -1,158 +1,216 @@
...
+2. Inspect diffs carefully and justify them in your PR description.
+3. Coordinate with maintainers when changes could affect downstream providers.
 
+## Debugging Failures
 
-## GRPC Replay Tests
+- Set `PULUMI_DEBUG_GRPC=1` for verbose Pulumi provider logs.
+- Enable Terraform tracing with `TF_LOG=DEBUG` or `TF_LOG=TRACE` when Terraform behavior is surprising.
+- Capture temporary files via `pulcheck.WithTempDir` helpers to inspect Terraform state or HCL.
+- For cross-tests, compare witness recordings under `pkg/internal/tests/cross-tests/witness`.
```
Comment (VenelinMartinov) [inline docs/guides/testing.md]: hallucination?

File: docs/guides/testing.md [hunk 3]
```diff
@@ -1,158 +1,216 @@
...
+1. Run `make test_accept` (sets `PULUMI_ACCEPT=1`) to update goldens.
+2. Inspect diffs carefully and justify them in your PR description.
+3. Coordinate with maintainers when changes could affect downstream providers.
 
+## Debugging Failures
 
-## GRPC Replay Tests
+- Set `PULUMI_DEBUG_GRPC=1` for verbose Pulumi provider logs.
+- Enable Terraform tracing with `TF_LOG=DEBUG` or `TF_LOG=TRACE` when Terraform behavior is surprising.
+- Capture temporary files via `pulcheck.WithTempDir` helpers to inspect Terraform state or HCL.
```
Comment (VenelinMartinov) [inline docs/guides/testing.md]: Hallucination?

File: docs/guides/testing.md [hunk 4]
```diff
@@ -1,158 +1,216 @@
...
+- For cross-tests, compare witness recordings under `pkg/internal/tests/cross-tests/witness`.
 
-These are discouraged. New tests should prefer [full schema and program tests](##-integration-tests-using-a-terraform-schema-and-a-pulumi-yaml-program)
+## Continuous Integration
 
-The bridge and bridged providers also supports GRPC replay tests. These tests are useful for testing the bridge against a specific GRPC request/response pair. 
-Examples can be found under [`./pkg/tfbridge/tests/provider_test.go`](https://github.com/pulumi/pulumi-terraform-bridge/blob/317d4b819f12d4bc66adc5fb248bfb77a3cc7ba7/pkg/tfbridge/provider_test.go#L190). These use the [`providertest.replay` module](https://github.com/pulumi/providertest/tree/main/replay) to replay a specific request/response pair from the Engine <-> Provider GRPC conversation.
\ No newline at end of file
+- GitHub Actions run `make lint` and `make test` with caching.
+- Longer suites (cross-tests, property tests) can be flaky; reproduce locally before retrying CI.
```
Comment (VenelinMartinov) [inline docs/guides/testing.md]: Is this true? AFAIK bridge tests aren't flaky

File: docs/guides/contributor-onboarding.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+Optional but recommended:
+
+- `mise` (see `mise.toml`) to manage tool versions.
+- Docker (some tests rely on Dockerized providers; note restrictions in `pkg/tests` if you run into issues).
+
+## Getting Set Up
+
+1. Clone the repo and optionally run `make go.work` to set up a Go workspace that includes nested modules.
+2. Run `make build` once; it tidies modules and ensures packages compile.
+3. Run `make test RUN_TEST_CMD=./pkg/tfbridge -run TestSmoke` (or another targeted test) to validate your local
```
Comment (VenelinMartinov) [inline docs/guides/contributor-onboarding.md]: Does this test exist?

File: docs/guides/contributor-onboarding.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+   environment before tackling larger changes. Full `make test` is heavy (downloads plugins and runs integration suites).
+
+## Repository Tour
+
+- **Runtime bridge** – `pkg/tfbridge`, `pkg/tfshim`: Pulumi RPC handlers + Terraform shims.
+- **Plugin Framework support** – `pkg/pf`, `pkg/x/muxer`.
```
Comment (VenelinMartinov) [inline docs/guides/contributor-onboarding.md]: The muxer isn't necessarily related to PF

File: docs/guides/contributor-onboarding.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+- **Design-time tooling** – `pkg/tfgen`, `pkg/convert`, `pkg/tf2pulumi`.
```
Comment (VenelinMartinov) [inline docs/guides/contributor-onboarding.md]: What is `design-time`?

File: docs/guides/contributor-onboarding.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+- **Dynamic provider** – `dynamic/` binary that parameterizes arbitrary Terraform providers.
```
Comment (VenelinMartinov) [inline docs/guides/contributor-onboarding.md]: Probably don't call it `dynamic provider` - I think that term is overloaded in pulumi. This should be `dynamically bridged provider`

File: docs/guides/contributor-onboarding.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+| Update shims | Touch `pkg/tfshim/sdk-v2/*` (or PF equivalents), run `make test RUN_TEST_CMD=./pkg/tfshim/sdk-v2`. |
+| Modify TFGen output | Edit `pkg/tfgen/*`, verify with unit tests and golden fixtures (`make test RUN_TEST_CMD=./pkg/tfgen`). |
+| Improve docs | Update Markdown in `docs/` and re-run Sphinx locally (`cd docs && make html`) if changing the rendered site. |
+| Accept golden updates | After intentionally changing output, run `make test_accept` and review diffs carefully. |
+
+## Debugging Tips
+
+- **Pulumi <-> Provider traffic** – Use `PULUMI_DEBUG_GRPC=1` during tests to inspect gRPC payloads.
+- **Terraform behavior** – Reproduce using Terraform CLI or cross-tests to confirm parity gaps.
+- **Schema diffs** – Leverage `pkg/tfgen/difftest` utilities and golden outputs.
```
Comment (VenelinMartinov) [inline docs/guides/contributor-onboarding.md]: Don't think this exists

File: docs/guides/contributor-onboarding.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+- **Panics / crashes** – Wrap providers with `providerserver.PanicRecoveringProvider` (already done in production) and
```
Comment (VenelinMartinov) [inline docs/guides/contributor-onboarding.md]: Don't think this is real advice

**Comment by corymhall** [comment]
> what are you looking for in terms of review? thoroughly or stamp? Can you highlight areas to double-check?

@t0yv0 good question! I'm trying to add/update some documentation to help the team as we ramp up on the bridge, but part of it is hard because I don't know what I don't know. 

For example, the testing piece has been one of the big confusing pieces to me. I tried to add some recommendations for what kind of tests to write, but I think it is still lacking. Am I correct in the way that I categorized it? If we deleted all the tests and started over would we only write the `pkg/tests` and `crosstests`?

Also, if you have any recommendations on the next items that we should focus on documenting I can add some backlog items for filling those out.

File: docs/architecture/overview.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
+# Pulumi Terraform Bridge Architecture Overview
+
+This document captures the high–level mental model maintainers use to reason about the Pulumi Terraform Bridge. It is
+intended as a launchpad: each section links to the code, docs, or playbooks that provide deeper detail.
+
+## System Goals
+
+The bridge has two equally important responsibilities:
+
+1. **Design-time generation** – Translate a Terraform provider's schema and metadata into a Pulumi schema, SDKs, and
```
Comment (VenelinMartinov) [inline docs/architecture/overview.md]: `Design-time generation`? Should this be `compile-time` maybe?

File: docs/architecture/overview.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+      │                 └─► Docs + Examples (`pkg/tfgen/docs.go`, `pkg/convert`)
+      ▼
+Pulumi SDK + Schema Artifacts
+```
+
+Key extension points:
+
+- **ProviderInfo overlays** (`pkg/tfbridge/info`) describe renames, doc overrides, examples, and custom CRUD behavior.
+- **Edit rules** (`pkg/tfgen/edit_rules.go`) post-process generated schema and docs.
+- **Converters** (`pkg/convert/*`) translate Terraform values to Pulumi `PropertyValue`s.
```
Comment (VenelinMartinov) [inline docs/architecture/overview.md]: no? I don't think this one is `design-time`

File: docs/architecture/overview.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+### Common Maintenance Tasks
+
+- Adding a new Terraform resource → update ProviderInfo + rerun `tfgen` via provider repo.
+- Debugging doc generation → reproduce with `make test RUN_TEST_CMD=./pkg/tfgen` and inspect `COVERAGE_OUTPUT_DIR` output.
+- Investigating schema deltas → use `pkg/tfgen/diff` utilities and golden tests.
```
Comment (VenelinMartinov) [inline docs/architecture/overview.md]: Does `pkg/tfgen/diff` exist?

File: docs/architecture/overview.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+
+## Runtime Pipeline
+
+```
+Pulumi Engine ─► `pkg/tfbridge.Provider` ─► `pkg/tfshim` ─► Terraform Provider
```
Comment (VenelinMartinov) [inline docs/architecture/overview.md]: This one doesn't account for the PF bridge - might be worth having separate diagrams for the two

File: docs/architecture/overview.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+  `pkg/tfbridge/provider_*.go` with shared helpers in `pkg/tfbridge/diff.go`, `pkg/tfbridge/schema.go`, etc.
+- **Shim layers** – `pkg/tfshim/sdk-v2` (and legacy `sdk-v1`) normalize Terraform Plugin SDK APIs; PF runtime uses
+  `pkg/pf/tfbridge` alongside `pkg/tfshim/schema`.
+- **Muxing / Migration** – `pkg/x/muxer` combines SDKv2 + PF implementations during migrations.
+- **Panic recovery** – `pkg/providerserver/panic_recovering_provider.go` guards against provider panics.
+
+### Cross-Cutting Concerns
+
+- **State translation** (`pkg/tfbridge/state.go`, `pkg/tfshim/sdk-v2/upgrade_state.go`).
+- **Diff & plan semantics** (`pkg/tfbridge/diff.go`, `pkg/pf/tfbridge/provider_diff.go`).
```
Comment (VenelinMartinov) [inline docs/architecture/overview.md]: Diff and state are very much runtime concerns

File: docs/architecture/overview.md [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+| **Static PF Bridge** | Providers authored with Plugin Framework. | `pkg/pf/tfbridge`, `pkg/pf/provider.go`. |
+| **Muxed Bridge** | Hybrid providers migrating resource-by-resource. | `pkg/x/muxer`, guides under `docs/guides/upgrade-sdk-to-mux.md`. |
+| **Dynamic Bridge** | Parameterize any registry provider at runtime. | `dynamic/main.go`, `dynamic/internal/shim`. |
+
+Each mode shares testing harnesses but has mode-specific fixtures (see `docs/guides/testing.md`).
+
+## Operational Flows
+
+- **Build & test** – `make build`, `make lint`, `make test`. See `AGENTS.md` and `docs/guides/contributor-onboarding.md`.
+- **Plugin pinning** – `make pin_upstream_sdk` keeps SDKv2 shims aligned with Pulumi fork.
```
Comment (VenelinMartinov) [inline docs/architecture/overview.md]: no - I think the plugin sdk fork prob deserves its own docs if you want to talk about that - this isn't right

**Review by VenelinMartinov — COMMENTED** [review]
I've meant to add some documentation but never got around to it! Thanks for tackling that!

I think there's some instances of hallucinations and I feel like the distinction between the PF and SDKv2 should be a bit more prominent but overall seems useful.

**Review by guineveresaenger — COMMENTED** [review]
The testing.md changes have the same issue I brought up here: https://github.com/pulumi/pulumi-terraform-bridge/pull/3192#discussion_r2380059572

Additionally, I feel that the information for pf tests has degraded here.

File: docs/guides/testing.md (hunk 4 above)
Comment (corymhall) [reply docs/guides/testing.md]: I think it may be basing it off of this comment in rapid https://github.com/pulumi/pulumi-terraform-bridge/blob/f865647b85ed81c2d7fdcefed1780f25861130d0/pkg/internal/tests/cross-tests/rapid_test.go#L33-L33

Do you know what PULUMI_EXPERIMENTAL is used for? It looks like it is used in a couple of places, but not in CI

https://github.com/search?q=repo%3Apulumi%2Fpulumi-terraform-bridge%20PULUMI_EXPERIMENTAL&type=code

File: docs/guides/testing.md (hunk 1 above)
Comment (corymhall) [reply docs/guides/testing.md:206]: I went ahead and updated the `make test_accept` target to also use `-update`. It looks like we still have both tests that use `PULUMI_ACCEPT` and tests that use autogold.

File: docs/guides/testing.md (hunk 4 above)
Comment (VenelinMartinov) [reply docs/guides/testing.md]: I suspect it was a "beta" flag at some point but now it's just the converter and the property-based tests

File: docs/guides/testing.md (hunk 1 above)
Comment (VenelinMartinov) [reply docs/guides/testing.md:206]: Adding `-update` will fail if you don't run a test executable which links in autogold - i.e. it'd trigger a failure for non-autogold tests. I think I had the same idea at some point but reverted after

Comment (corymhall) [reply docs/guides/testing.md:206]: ah ok, i'll just update this doc to note that you have to individually update autogold tests

File: docs/guides/testing.md (hunk 4 above)
Comment (corymhall) [reply docs/guides/testing.md]: When do we run those? Should developers be using that flag for certain things? Should we remove it or turn it on in CI?

File: docs/architecture/overview.md (hunk 12 above)
Comment (corymhall) [reply docs/architecture/overview.md]: If I `/s/design-time/compile-time/` would that make sense? Other options:

- build-time
- generation-time
- ...

File: docs/architecture/overview.md (hunk 13 above)
Comment (corymhall) [reply docs/architecture/overview.md]: It hallucinated this in both docs, weird.

File: docs/guides/testing.md (hunk 4 above)
Comment (VenelinMartinov) [reply docs/guides/testing.md]: The property-based tests were intended as more of a local tool, we never got them to a stable place where they can run in CI without failures

Comment (corymhall) [reply docs/guides/testing.md]: Ok I added a note about this.

File: docs/architecture/overview.md (hunk 12 above)
Comment (VenelinMartinov) [reply docs/architecture/overview.md]: Yeah, perhaps build-time makes more sense here? It's doing more than just compilation

File: docs/architecture/overview.md (hunk 14 above)
Comment (corymhall) [reply docs/architecture/overview.md]: I separated it into two sections

File: docs/guides/contributor-onboarding.md (hunk 6 above)
Comment (corymhall) [reply docs/guides/contributor-onboarding.md]: ok broke that out

File: docs/architecture/overview.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
+# Pulumi Terraform Bridge Architecture Overview
+
+This document captures the high–level mental model maintainers use to reason about the Pulumi Terraform Bridge. It is
+intended as a launchpad: each section links to the code, docs, or playbooks that provide deeper detail.
+
+## System Goals
+
+The bridge has two equally important responsibilities:
+
+1. **Build-time generation** – Translate a Terraform provider's schema and metadata into a Pulumi schema, SDKs, and docs so Pulumi users can program against the provider.
```
Comment (t0yv0) [inline docs/architecture/overview.md]: SDKs are not the responsibility of the bridge. This is worth making very clear. SDKs are generated from the Pulumi Package schema as a source of truth. Worth linking https://www.pulumi.com/docs/iac/build-with-pulumi/schema/ perhaps.

File: docs/architecture/overview.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+2. **Runtime translation** – Act as a Pulumi resource provider that speaks Pulumi's gRPC protocol while driving the
+   Terraform provider implementation under the hood.
+
+A change to either half can impact the other. Treat the bridge as one product with two tightly-coupled pipelines.
+
+## Major Components
+
+| Layer | Location | Role |
+| ----- | -------- | ---- |
+| Build-time | `pkg/tfgen`, `docs/guides/*`, `pkg/convert`, `pkg/tf2pulumi` | Introspect Terraform provider schemas, emit Pulumi schema, SDKs, and docs. |
```
Comment (t0yv0) [inline docs/architecture/overview.md]: tf2pulumi needs to be removed. It's been straggling but it does not belong here.  Worth documenting accordingly so it does not get more confusing.

File: docs/architecture/overview.md [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+      │                 └─► Docs + Examples (`pkg/tfgen/docs.go`, `pkg/convert`)
+      ▼
+Pulumi SDK + Schema Artifacts
+```
+
+Key extension points:
+
+- **ProviderInfo overlays** (`pkg/tfbridge/info`) describe renames, doc overrides, examples, and custom CRUD behavior.
+- **Edit rules** (`pkg/tfgen/edit_rules.go`) post-process generated schema and docs.
+- **Converters** (`pkg/convert/*`) translate Terraform values to Pulumi `PropertyValue`s.
```
Comment (t0yv0) [inline docs/architecture/overview.md]: pkg/convert is not important.. I would not highlight it here... I'd use package.go to document what the package is about.

File: docs/architecture/overview.md [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+- **Lifecycle RPCs** – `Create`, `Read`, `Update`, `Delete`, `Invoke`, `Plan`, `CheckConfig`, etc. live in
+  `pkg/tfbridge/provider_*.go` with shared helpers in `pkg/tfbridge/diff.go`, `pkg/tfbridge/schema.go`, etc.
+- **Shim layers** – `pkg/tfshim/sdk-v{1,2}` normalize Terraform Plugin SDK APIs into bridge-friendly interfaces.
+- **Panic recovery** – `pkg/providerserver/panic_recovering_provider.go` guards against provider panics.
+- **Diff & plan semantics** – core logic in `pkg/tfbridge/diff.go`, `pkg/tfbridge/detailed_diff.go`, and related helpers.
+
+## Runtime Pipeline (Plugin Framework)
+
+```
+Pulumi Engine ─► `pkg/pf/tfbridge.Provider` ─► `pf.ShimProvider` / `pkg/pf/internal/schemashim` ─► `tfprotov6.ProviderServer` ─► Terraform Plugin Framework provider
```
Comment (t0yv0) [inline docs/architecture/overview.md]: schemashim is not used at runtime :/

File: docs/architecture/overview.md [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+
+### Common Maintenance Tasks
+
+- Adding a new Terraform resource → update ProviderInfo + rerun `tfgen` via provider repo.
+- Debugging doc generation → reproduce with `make test RUN_TEST_CMD=./pkg/tfgen` and inspect `COVERAGE_OUTPUT_DIR` output.
+
+## Runtime Pipeline (Plugin SDK v2)
+
+```
+Pulumi Engine ─► `pkg/tfbridge.Provider` ─► `pkg/tfshim/sdk-v2` ─► Terraform Plugin SDK provider
```
Comment (t0yv0) [inline docs/architecture/overview.md:47]: This is somewhat helpful.

File: docs/guides/testing.md [hunk 22]
```diff
@@ -1,158 +1,218 @@
...
+- Create/Update flows (`pkg/tfbridge/tests/provider_test.go`)
+- Provider configuration (`pkg/tfbridge/tests/provider_configure_test.go`, `pkg/pf/tests/provider_configure_test.go`)
+- PF-specific diffing (`pkg/pf/tests/diff_test.go`)
+
+## Property-Based Tests
+
+Property-based tests extend cross-tests with randomized inputs using the
+[Rapid](https://github.com/flyingmutant/rapid) library (`pkg/internal/tests/cross-tests/rapid_test.go`). These tests
+are currently intended to be used as a local tool. They are not yet in a stable place where
+we can run these in CI without failures.
```
Comment (t0yv0) [inline docs/guides/testing.md:194]: Is this true?

File: docs/guides/testing.md [hunk 23]
```diff
@@ -1,158 +1,218 @@
...
+```
+
+`make test` installs required Pulumi plugins and builds the test provider binary, so the first run can take several minutes.
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tests/pulcheck"
-)
+## Choosing the Right Layer
 
+1. **Start simple** – If logic can be isolated, write a pure Go unit test.
```
Comment (t0yv0) [inline docs/guides/testing.md]: This is a great conversation starter, but do we believe this? 

In my mind the selection criteria are like this.

First on individual tests vs randomized test matrices via rapid:

- regression tests call for individual test cases being added 
- proactive hardening work to suss out edge cases calls for rapid/PBT; this is only appropriate for super stable core of the system

Second on using cross-tests vs testing against Pulumi only:
- use cross tests when the behavior target is matching TF - the doc gets this right "Write or update a cross-test so Terraform CLI and Pulumi remain aligned." but it's not limited to regressions, supporting new TF features for example is also appropriate here
- use regular tests against Pulumi CLI for behavior that is by-design specific to Pulumi

Simple Go unit tests are not as useful for this codebase as correctness here is defined by translating Pulumi semantics to TF semantics; thankfully with the test infrastructure in-place these integration tests are pretty fast to run.

File: docs/guides/testing.md (hunk 4 above)
Comment (t0yv0) [reply docs/guides/testing.md]: It's not that its' flaky, it's they're finding unfixed bugs maybe. That's what the comment meant I think.

>  The property-based tests were intended as more of a local tool

I do not think that is quite true in how they were intended. I think the team just gave up on that extra level of correctness because it takes a lot of work to get there. I don't think the problem is going away though - as more and more stars and users are added to providers in https://github.com/pulumi/pulumi-aws it becomes more and more important to meet a higher quality bar, and IME without techniques like fuzz testing it's just not going to happen.

**Comment by t0yv0** [comment]
Did a bit of review, ty! AI is doing a good job given the context it has but we could do better talking and writing more maybe? In a sense the higher-level confusion is the most costly confusion. IDK, thanks for prompting this work anyway! Perhaps a few rounds of AI -> edits is what we need to move forward. Appreciate the ping!

File: docs/guides/testing.md (hunk 4 above)
Comment (corymhall) [reply docs/guides/testing.md]: Do we instead need a tracking issue to link and say it's a WIP?

File: docs/guides/testing.md (hunk 22 above)
Comment (corymhall) [reply docs/guides/testing.md:194]: It's at least true that we _don't_ run them in CI.

File: docs/guides/testing.md (hunk 4 above)
Comment (t0yv0) [reply docs/guides/testing.md]: Is it WIP? The team needs to decide if it's WIP or abandoned. I'm just you my tech takes from before but since I'm not working on this codebase for a bit those are outside opinions.

File: docs/guides/testing.md (hunk 22 above)
Comment (t0yv0) [reply docs/guides/testing.md:194]: Ah yeah perhaps it is true! Thanks! Just checking.

File: docs/architecture/overview.md (hunk 18 above)
Comment (iwahbe) [inline docs/architecture/overview.md]: Putting `docs/guides` here is a bit funny. I don't think of docs as a build time concern.

File: docs/architecture/overview.md [hunk 24]
```diff
@@ -0,0 +1,115 @@
...
+
+A change to either half can impact the other. Treat the bridge as one product with two tightly-coupled pipelines.
+
+## Major Components
+
+| Layer | Location | Role |
+| ----- | -------- | ---- |
+| Build-time | `pkg/tfgen`, `docs/guides/*`, `pkg/convert`, `pkg/tf2pulumi` | Introspect Terraform provider schemas, emit Pulumi schema, SDKs, and docs. |
+| Runtime (SDKv2) | `pkg/tfbridge`, `pkg/tfshim/sdk-v{1,2}`, `pkg/providerserver` | Drive Terraform Plugin SDK providers via Pulumi RPC entry points. |
+| Runtime (PF) | `pkg/pf/*`, `pkg/tfshim/schema` | Bridge Terraform Plugin Framework providers. |
```
Comment (iwahbe) [inline docs/architecture/overview.md]:
```suggestion
| Runtime (PF) | `pkg/pf/tfbridge`, `pkg/tfshim/schema` | Bridge Terraform Plugin Framework providers. |
```

File: docs/architecture/overview.md (hunk 18 above)
Comment (iwahbe) [inline docs/architecture/overview.md]:
```suggestion
| Build-time | `pkg/tfgen`, `pkg/pf/tfgen`, `docs/guides/*`, `pkg/convert`, `pkg/tf2pulumi` | Introspect Terraform provider schemas, emit Pulumi schema, SDKs, and docs. |
```

File: docs/architecture/overview.md [hunk 25]
```diff
@@ -0,0 +1,115 @@
...
+| Hybrid / muxing | `pkg/x/muxer`, `docs/guides/upgrade-sdk-to-mux.md` | Compose multiple runtime pipelines (SDKv2, PF, dynamic) into a single provider surface. |
+| Dynamic bridge | `dynamic/*` | Parameterizable provider that downloads and hosts Terraform providers at runtime. |
```
Comment (iwahbe) [inline docs/architecture/overview.md]: This is effectively a separate project compared to the rest of the bridge. It is versioned and released separately, and is never depended upon by static bridge code.

File: docs/architecture/overview.md [hunk 26]
```diff
@@ -0,0 +1,115 @@
...
+| Testing | `pkg/tests`, `pkg/internal/tests/cross-tests`, `pkg/pf/tests` | Harnesses that keep behavior aligned with Terraform and Pulumi expectations. |
+| Ops & Tooling | `Makefile`, `scripts/`, `docs/operations` | Build, lint, test, release, and operational playbooks. |
+
+## Build-Time Pipeline
+
+```
+Terraform Provider ──► Schema Introspection (`tfgen/schema.go`)
+      │                 │
+      │                 ├─► Pulumi PackageSpec (`pkg/tfgen/generate.go`)
+      │                 ├─► Language SDKs (`pkg/tfgen/generate_<lang>.go`)
```
Comment (iwahbe) [inline docs/architecture/overview.md]:
```suggestion
      │                 ├─► Pulumi Schema (`pkg/tfgen/generate.go`)
```

File: docs/architecture/overview.md [hunk 27]
```diff
@@ -0,0 +1,115 @@
...
+
+- **ProviderInfo overlays** (`pkg/tfbridge/info`) describe renames, doc overrides, examples, and custom CRUD behavior.
+- **Edit rules** (`pkg/tfgen/edit_rules.go`) post-process generated schema and docs.
+- **Converters** (`pkg/convert/*`) translate Terraform values to Pulumi `PropertyValue`s.
+
+### Common Maintenance Tasks
+
+- Adding a new Terraform resource → update ProviderInfo + rerun `tfgen` via provider repo.
+- Debugging doc generation → reproduce with `make test RUN_TEST_CMD=./pkg/tfgen` and inspect `COVERAGE_OUTPUT_DIR` output.
+
```
Comment (iwahbe) [inline docs/architecture/overview.md]: Are we talking about maintenance tasks for people who maintain providers or people who maintain the bridge?

File: docs/architecture/overview.md [hunk 28]
```diff
@@ -0,0 +1,115 @@
...
+- **Build & test** – `make build`, `make lint`, `make test`. See `AGENTS.md` and `docs/guides/contributor-onboarding.md`.
+- **Releases** – Follow platform providers playbook (link in root README); capture bridge-specific steps in
+  `docs/operations/maintainability-playbook.md`.
+
+## Where to Learn More
+
+- Runtime deep dive (planned) → `docs/architecture/runtime.md` (placeholder).
+- Build-time cookbook → `docs/guides/new-provider.md`, `docs/guides/new-pf-provider.md`.
+- Testing strategy → `docs/guides/testing.md` (this repo) + `pkg/tests` examples.
+- Maintaining TODOs & deprecations → `docs/operations/todo-triage.md`.
```
Comment (iwahbe) [inline docs/architecture/overview.md]: All links like this should be relative MD links that can be clicked through. That applies to this whole document.

File: docs/guides/contributor-onboarding.md [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+
+Welcome! This guide distills the minimum context required to make your first change to the Pulumi Terraform Bridge. It
+assumes you are comfortable with Go and have Pulumi/Terraform installed, but are new to this repository.
+
+## Prerequisites
+
+- Go version from `go.mod` (install via `mise` or your package manager).
+- Pulumi CLI ≥ 3.x (required for tests and plugin installs).
+- Terraform / OpenTofu CLI (helps when debugging or running cross-tests locally).
+- `golangci-lint` (installed automatically via `make lint`).
```
Comment (iwahbe) [inline docs/guides/contributor-onboarding.md]: If it can be installed automatically (and it always will be), it's not a prerequisite.
```suggestion
```

File: docs/guides/contributor-onboarding.md [hunk 30]
```diff
@@ -0,0 +1,63 @@
...
+
+## Debugging Tips
+
+- **Pulumi <-> Provider traffic** – Use `PULUMI_DEBUG_GRPC=1` during tests to inspect gRPC payloads.
+- **Terraform behavior** – Reproduce using Terraform CLI or cross-tests to confirm parity gaps.
+
+## Coding Standards
+
+- Run `make fmt` (gofmt -s) and `make lint` (golangci-lint) before sending a PR.
+- Favor explicit module paths under `github.com/pulumi/pulumi-terraform-bridge/v3/...`.
```
Comment (iwahbe) [inline docs/guides/contributor-onboarding.md]: This doesn't mean anything. Import paths will be correct.

File: docs/architecture/overview.md (hunk 25 above)
Comment (corymhall) [reply docs/architecture/overview.md]: Should I just note that and point to that project's README?

File: docs/architecture/overview.md (hunk 27 above)
Comment (corymhall) [reply docs/architecture/overview.md]: Yeah this is out of place, i'll remove.

File: docs/guides/testing.md (hunk 23 above)
Comment (corymhall) [reply docs/guides/testing.md]: Ok I updated the wording here. Let me know what you think.

File: pkg/pf/README.md [hunk 31]
```diff
@@ -1,16 +1,12 @@
-# Pulumi Bridge for Terraform Plugin Framework
+# pkg/pf
 
-This bridge enables creating [Pulumi Resource providers](https://www.pulumi.com/docs/intro/concepts/resources/providers/) from [Terraform Providers](https://github.com/terraform-providers)
-built using the [Terraform Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework).
+`pkg/pf` contains the bridge implementation for Terraform Plugin Framework (PF) providers.
 
-If you need to adapt [Terraform Plugin SDK](https://github.com/hashicorp/terraform-plugin-sdk) based providers, see [the documentation for
-bridging a new SDK based provider](../../docs/guides/new-provider.md).
+- `provider.go` exposes helpers to construct Pulumi providers backed by PF implementations.
```
Comment (VenelinMartinov) [inline pkg/pf/README.md]: This seems like a hallucination - `provider.go` contains just an interface
